packages/mdx/lib/plugin/recma-jsx-rewrite.js

```diff
@@ -193,16 +195,21 @@ export function recmaJsxRewrite(options = {}) {
               fnScope.tags.push(id)
             }
 
-            node.openingElement.name = toJsxIdOrMemberExpression([
-              '_components',
-              id
-            ])
+            let componentName = toJsxIdOrMemberExpression(['_components', id])
+            if (isIdentifierName(id) === false) {
+              let invalidComponentName = invalidComponentNameToVariable.get(id)
+              if (invalidComponentName === undefined) {
+                invalidComponentName = `_component${invalidComponentNameToVariable.size}`
+                invalidComponentNameToVariable.set(id, invalidComponentName)
+              }
+
+              componentName = toJsxIdOrMemberExpression([invalidComponentName])
+            }
+
+            node.openingElement.name = componentName
 
             if (node.closingElement) {
-              node.closingElement.name = toJsxIdOrMemberExpression([
-                '_components',
-                id
-              ])
+              node.closingElement.name = componentName
```

Comment:
This needs to be cloned. It’s a weird edge case, but we shouldn’t have the same object multiple times in the tree.